docs/design/graceful-domain-failover/3051-graceful-domain-failover.md

```diff
@@ -0,0 +1,241 @@
+# Design doc: Cadence graceful domain failover
+
+Author: Cadence Team
+
+Last updated: Mar 2020
+
+Reference: [#3051](https://github.com/uber/cadence/issues/3051)
+
+
+## Abstract
+
+Cadence supports domain failover with multi-cluster setup. However, the problems with the current failover are:
+1. Workflow progress could be lost.
+2. No causal consistency guarantee on domain level.
+
+The graceful domain failover uses to solve those two problems.
+
+## Use cases
+
+Users can trigger graceful domain failover via Cadence CLI:
+
+`cadence --domain cadence-global domain failover graceful -active_cluster XYZ -failover_timeout 120s`
+
+Users can force complete domain failover via Cadence CLI:
+
+`cadence --domain cadence-global domain update force -active_cluster XYZ`
```

Comment:
It is a little bit confusing that one is called domain failover and the other is called domain update. We should make it consistent.

Reply:
update the CLI to be consistent